src/ParallelAlgorithms/LinearSolver/Schwarz/ComputeTags.hpp

```diff
@@ -0,0 +1,177 @@
+      tmpl::list<domain::Tags::Mesh<Dim>, domain::Tags::Element<Dim>,
+                 IntrudingExtents<Dim, OptionsGroup>,
+                 domain::Tags::Direction<Dim>,
+                 domain::Tags::Coordinates<Dim, Frame::Logical>,
+                 IntrudingOverlapWidths<Dim, OptionsGroup>>;
+  using volume_tags =
+      tmpl::list<domain::Tags::Mesh<Dim>, domain::Tags::Element<Dim>,
+                 IntrudingExtents<Dim, OptionsGroup>,
+                 domain::Tags::Coordinates<Dim, Frame::Logical>,
+                 IntrudingOverlapWidths<Dim, OptionsGroup>>;
+  static constexpr void function(
+      const gsl::not_null<return_type*> intruding_overlap_weight,
+      const Mesh<Dim>& mesh, const Element<Dim>& element,
+      const std::array<size_t, Dim>& intruding_extents,
+      const Direction<Dim>& direction,
+      const tnsr::I<DataVector, Dim, Frame::Logical>& logical_coords,
+      const std::array<double, Dim> intruding_overlap_widths) noexcept {
+    const size_t dim = direction.dimension();
+    if (gsl::at(intruding_extents, dim) > 0) {
+      const auto intruding_logical_coords =
+          LinearSolver::Schwarz::data_on_overlap(
+              logical_coords, mesh.extents(), gsl::at(intruding_extents, dim),
+              direction);
+      LinearSolver::Schwarz::intruding_weight(
+          intruding_overlap_weight, intruding_logical_coords, direction,
+          intruding_overlap_widths, element.neighbors().at(direction).size(),
+          element.external_boundaries());
+    }
+  }
+};
+
+/*!
+ * \brief A diagnostic quantity to check that weights are conserved
+ *
+ * \see `LinearSolver::Schwarz::Tags::SummedIntrudingOverlapWeights`
+ */
+template <size_t Dim, typename OptionsGroup>
+struct SummedIntrudingOverlapWeightsCompute
+    : db::ComputeTag,
+      SummedIntrudingOverlapWeights<OptionsGroup> {
+  using base = SummedIntrudingOverlapWeights<OptionsGroup>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<domain::Tags::Interface<domain::Tags::InternalDirections<Dim>,
+                                         Weight<OptionsGroup>>,
+                 domain::Tags::Mesh<Dim>, IntrudingExtents<Dim, OptionsGroup>>;
+  static void function(
+      const gsl::not_null<return_type*> summed_intruding_overlap_weights,
+      const std::unordered_map<Direction<Dim>, Scalar<DataVector>>&
+          all_intruding_weights,
+      const Mesh<Dim>& mesh,
+      const std::array<size_t, Dim>& all_intruding_extents) noexcept {
+    destructive_resize_components(summed_intruding_overlap_weights,
+                                  mesh.number_of_grid_points());
+    get(*summed_intruding_overlap_weights) = 0.;
+    for (const auto& [direction, intruding_weight] : all_intruding_weights) {
+      // Extend intruding weight to full extents
+      // There's not function to extend a single tensor, so we create a
+      // temporary Variables. This is only a diagnostic quantity that won't be
```

Comment:
It's diagnostic in the sense that constraint violation observations are - it can be added to the DataBox of an executable to make sure the weighting works correctly, in particular when we have AMR.

Reply:
I'm not sure I understand the analogy to constraints -- this doesn't appear to be something that will depend on the physical variables as they are iterated. It seems more like an identity that should always hold to some desired precision, and should depend only on the functions that determine the weights. Am I misunderstanding?

Reply:
yea I deleted the mention of constraints after submitting the comment, it's not a good analogy.